Bottom: 8bbfd91bac2c1b1ab10dab89b91b51b1a1ba3848
Top:    3714a8d9b088e67f4ed134a3b75bd33449f34c0e
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-11-14 17:20:09 +0800

Refresh of tuntap-ndo_xdp_xmit-support

---

diff --git a/include/linux/if_tun.h b/include/linux/if_tun.h
index 31ce3b0..1cafdc2 100644
--- a/include/linux/if_tun.h
+++ b/include/linux/if_tun.h
@@ -25,7 +25,6 @@ struct ptr_ring *tun_get_tx_ring(struct file *file);
 bool tun_is_xdp_buff(void *ptr);
 void *tun_xdp_to_ptr(void *ptr);
 void *tun_ptr_to_xdp(void *ptr);
-
 #else
 #include <linux/err.h>
 #include <linux/errno.h>
@@ -39,5 +38,17 @@ static inline struct ptr_ring *tun_get_tx_ring(struct file *f)
 {
 	return ERR_PTR(-EINVAL);
 }
+static inline bool tun_is_xdp_buff(void *ptr)
+{
+	return false;
+}
+void *tun_xdp_to_ptr(void *ptr)
+{
+	return NULL;
+}
+void *tun_ptr_to_xdp(void *ptr)
+{
+	return NULL;
+}
 #endif /* CONFIG_TUN */
 #endif /* __IF_TUN_H */
